### PR TITLE
Updated the `execute_with_gradients` function and `README` examples for better usability

### PR DESCRIPTION
**Pull request by vedpatwardhan**
1. Changed the default value of `xs_grad_idxs` in `execute_with_gradients` to `[[0]]` when `xs` is not a `ivy.Array` or `ivy.Container`, as the most common usage would involve `xs` being `(variables, input, output)`.
2. Changed the default value of `ret_grad_idxs` in `execute_with_gradients` to `[[0]]` as well when `ret` from the `func` is not an `ivy.Array` or `ivy.Container`, for consistency with `xs_grad_idxs` and to make it more intuitive.
3. Updated the examples in the `README.rst` to having to pass these arguments for the most common use-case.
4. Moved hyperparameter initialization above model creation in the `README.rst` demo for image classification
5. Skipped the tests for `execute_with_gradients` with the `numpy` backend.

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by vedpatwardhan** [comment]
As the `README` was changed, could you please look through the PR @guillesanbri ? Thanks 🙂

File: README.rst [hunk 1]
```diff
@@ -294,7 +294,7 @@ but this can easily be changed to your favorite framework, such as TensorFlow, o
         pred = model(x)
 
         # compute loss and gradients
-        loss, grads = ivy.execute_with_gradients(lambda params: loss_fn(*params), (model.v, x, y), xs_grad_idxs=[[0]])
+        loss, grads = ivy.execute_with_gradients(lambda params: loss_fn(*params), (model.v, x, y))
 
         # update parameters
         model.v = optimizer.step(model.v, grads)
```
Comment (guillesanbri) [inline README.rst:300]: I'm getting this
```bash
    model.v = optimizer.step(model.v, grads)
  File "/workspace/graph-compiler/ivy_repo/ivy/stateful/optimizers.py", line 157, in step
    return self._step_fn(v, grads, ignore_missing)
  File "/workspace/graph-compiler/ivy_repo/ivy/stateful/optimizers.py", line 112, in _step_fn
    return self._step(v, grads)
  File "/workspace/graph-compiler/ivy_repo/ivy/stateful/optimizers.py", line 386, in _step
    self._vw = grads**2

TypeError: unsupported operand type(s) for ** or pow(): 'list' and 'int'
```
on latest master, would be good to look into this before merging :+1:

**Review by guillesanbri — CHANGES_REQUESTED** [review]

File: README.rst (hunk 1 above)
Comment (guillesanbri) [reply README.rst:300]: the rest looks good to me :+1:

Comment (vedpatwardhan) [reply README.rst:300]: No worries! I'll get the PR merged in that case, thanks @guillesanbri 😄

**Pull request merged**